app/controllers/automated_tests_controller.rb

```diff
@@ -220,4 +222,17 @@ def required_params
   def assignment_params
     params.require(:assignment).permit(*required_params)
   end
+
+  def find_or_create_test_student(assignment_id)
+    test_group = Group.find_or_create_by(group_name: 'test_student_group')
```

Comment:
Since the Group.group_name attribute has to be unique, reserve this group name by assigning it to a constant in the Group class and add a validation to make sure that no other group can have this group_name. Otherwise you might end up with a real group that conflicts with this group_name.

Do the same thing below for the TestStudent.user_name.